From e15578075bb2c13dc92683187afc12bd1949d1d1 Mon Sep 17 00:00:00 2001
From: helifu <helifu@gmail.com>
Date: Tue, 19 May 2020 09:08:15 +0800
Subject: [PATCH] Don't use MADV_FREE on Linux

---
 src/system-alloc.cc | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/system-alloc.cc b/src/system-alloc.cc
index 59cd03d..4cdae94 100755
--- a/src/system-alloc.cc
+++ b/src/system-alloc.cc
@@ -62,6 +62,18 @@
 # define MAP_ANONYMOUS MAP_ANON
 #endif

+// Linux added support for MADV_FREE in 4.5 but we aren't ready to use it
+// yet. Among other things, using compile-time detection leads to poor
+// results when compiling on a system with MADV_FREE and running on a
+// system without it. See https://github.com/gperftools/gperftools/issues/780.
+//
+// At the same time, we also reserve an method to enable usage of MADV_FREE
+// by building with -DTCMALLOC_USE_MADV_FREE which could help people to test
+// it more easily.
+#if defined(__linux__) && defined(MADV_FREE) && !defined(TCMALLOC_USE_MADV_FREE)
+# undef MADV_FREE
+#endif
+
 // MADV_FREE is specifically designed for use by malloc(), but only
 // FreeBSD supports it; in linux we fall back to the somewhat inferior
 // MADV_DONTNEED.

