diff -Naur opentelemetry-cpp-1.20.0/CMakeLists.txt opentelemetry-cpp-1.20.0.patched/CMakeLists.txt
--- opentelemetry-cpp-1.20.0/CMakeLists.txt	2025-04-01 15:21:41.000000000 -0700
+++ opentelemetry-cpp-1.20.0.patched/CMakeLists.txt	2026-10-08 09:05:20.727066018 -0700
@@ -37,7 +37,9 @@
 mark_as_advanced(CMAKE_TOOLCHAIN_FILE)
 
 # Prefer cmake CONFIG to auto resolve dependencies.
-set(CMAKE_FIND_PACKAGE_PREFER_CONFIG TRUE)
+if(NOT DEFINED CMAKE_FIND_PACKAGE_PREFER_CONFIG)
+  set(CMAKE_FIND_PACKAGE_PREFER_CONFIG TRUE)
+endif()
 
 # Don't use customized cmake modules if vcpkg is used to resolve dependence.
 if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
