From c1510831033868f6c6fd62ad2ef9d202d7adf514 Mon Sep 17 00:00:00 2001
From: Hasan Alayli <halayli@gmail.com>
Date: Sat, 16 Sep 2017 06:32:40 -0700
Subject: [PATCH] ORC-239. Install missing Statistics.hh header file.

Fixes #168

Signed-off-by: Owen O'Malley <omalley@apache.org>
---
 c++/include/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/c++/include/CMakeLists.txt b/c++/include/CMakeLists.txt
index 901d6b2..bd32f7b 100644
--- a/c++/include/CMakeLists.txt
+++ b/c++/include/CMakeLists.txt
@@ -80,5 +80,6 @@ install(FILES
   "orc/Reader.hh"
   "orc/Type.hh"
   "orc/Vector.hh"
+  "orc/Statistics.hh"
   DESTINATION "include/orc"
   )
-- 
2.7.4

