--- breakpad-1b704857f1e78a864e6942e613457e55f1aecb60-p2.orig/src/tools/linux/md2core/minidump-2-core.cc	2017-08-11 10:53:27.875103632 -0700
+++ breakpad-1b704857f1e78a864e6942e613457e55f1aecb60-p2/src/tools/linux/md2core/minidump-2-core.cc	2017-08-11 10:55:09.640909794 -0700
@@ -963,16 +963,6 @@
     return false;
   }
 
-#if !defined(__PPC__)
-  nhdr.n_descsz = sizeof(user_fpregs_struct);
-  nhdr.n_type = NT_FPREGSET;
-  if (!writea(1, &nhdr, sizeof(nhdr)) ||
-      !writea(1, "CORE\0\0\0\0", 8) ||
-      !writea(1, &thread.fpregs, sizeof(user_fpregs_struct))) {
-    return false;
-  }
-#endif
-
 #if defined(__i386__) || defined(__x86_64__)
   nhdr.n_descsz = sizeof(user_fpregs_struct);
   nhdr.n_type = NT_FPREGSET;
