--- bzip2-1.0.6/Makefile	2010-09-10 15:46:02.000000000 -0700
+++ bzip2-1.0.6/Makefile.new	2015-06-02 08:26:24.470361539 -0700
@@ -5,7 +5,7 @@
 # bzip2/libbzip2 version 1.0.6 of 6 September 2010
 # Copyright (C) 1996-2010 Julian Seward <jseward@bzip.org>
 #
-# Please read the WARNING, DISCLAIMER and PATENTS sections in the 
+# Please read the WARNING, DISCLAIMER and PATENTS sections in the
 # README file.
 #
 # This program is released under the terms of the license contained
@@ -15,13 +15,13 @@
 SHELL=/bin/sh
 
 # To assist in cross-compiling
-CC=gcc
-AR=ar
-RANLIB=ranlib
-LDFLAGS=
+CC ?= gcc
+AR ?= ar
+RANLIB ?= ranlib
+LDFLAGS ?=
 
-BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-Wall -Winline -O2 -g $(BIGFILES)
+BIGFILES = -D_FILE_OFFSET_BITS=64
+CFLAGS += -Wall -Winline -O2 -g $(BIGFILES)
 
 # Where you want it installed when you do 'make install'
 PREFIX=/usr/local
@@ -61,7 +61,7 @@
 	./bzip2 -d  < sample1.bz2 > sample1.tst
 	./bzip2 -d  < sample2.bz2 > sample2.tst
 	./bzip2 -ds < sample3.bz2 > sample3.tst
-	cmp sample1.bz2 sample1.rb2 
+	cmp sample1.bz2 sample1.rb2
 	cmp sample2.bz2 sample2.rb2
 	cmp sample3.bz2 sample3.rb2
 	cmp sample1.tst sample1.ref
@@ -108,7 +108,7 @@
 	echo ".so man1/bzmore.1" > $(PREFIX)/man/man1/bzless.1
 	echo ".so man1/bzdiff.1" > $(PREFIX)/man/man1/bzcmp.1
 
-clean: 
+clean:
 	rm -f *.o libbz2.a bzip2 bzip2recover \
 	sample1.rb2 sample2.rb2 sample3.rb2 \
 	sample1.tst sample2.tst sample3.tst
@@ -203,7 +203,7 @@
 # For rebuilding the manual from sources on my SuSE 9.1 box
 
 MANUAL_SRCS= 	bz-common.xsl bz-fo.xsl bz-html.xsl bzip.css \
-		entities.xml manual.xml 
+		entities.xml manual.xml
 
 manual: manual.html manual.ps manual.pdf
 
