From 8e0bf8d631b79dc2ab095371f8530109c8749a43 Mon Sep 17 00:00:00 2001
From: Daniel Becker <dbecker@apache.org>
Date: Mon, 19 May 2025 10:49:02 +0200
Subject: [PATCH] Make glog internals visible.

---
 CMakeLists.txt | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 846b444..5c5d42e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,10 +60,7 @@ if (NOT WITH_THREADS)
   set (CMAKE_DISABLE_FIND_PACKAGE_Threads ON)
 endif (NOT WITH_THREADS)
 
-set (CMAKE_C_VISIBILITY_PRESET hidden)
-set (CMAKE_CXX_VISIBILITY_PRESET hidden)
 set (CMAKE_POSITION_INDEPENDENT_CODE ON)
-set (CMAKE_VISIBILITY_INLINES_HIDDEN ON)
 
 set (CMAKE_DEBUG_POSTFIX d)
 set (CMAKE_THREAD_PREFER_PTHREAD 1)
@@ -581,6 +578,7 @@ set (GLOG_PUBLIC_H
   ${CMAKE_CURRENT_BINARY_DIR}/glog/vlog_is_on.h
   src/glog/log_severity.h
   src/glog/platform.h
+  src/utilities.h
 )
 
 set (GLOG_SRCS
-- 
2.34.1

