diff -Naur thrift-0.16.0/lib/cpp/src/thrift/concurrency/Mutex.h thrift-0.16.0.patched/lib/cpp/src/thrift/concurrency/Mutex.h
--- thrift-0.16.0/lib/cpp/src/thrift/concurrency/Mutex.h	2021-08-03 13:11:21.000000000 -0700
+++ thrift-0.16.0.patched/lib/cpp/src/thrift/concurrency/Mutex.h	2025-10-31 17:10:40.380241803 -0700
@@ -20,6 +20,7 @@
 #ifndef _THRIFT_CONCURRENCY_MUTEX_H_
 #define _THRIFT_CONCURRENCY_MUTEX_H_ 1
 
+#include <cstdint>
 #include <memory>
 #include <thrift/TNonCopyable.h>
 
