From 9929fcb253228ef38d96e14ca8f598e77a80783e Mon Sep 17 00:00:00 2001
From: Nobuaki Sukegawa <nsukeg@gmail.com>
Date: Tue, 23 Dec 2014 19:24:28 +0900
Subject: [PATCH] THRIFT-2910 libevent is not really optional

---
 lib/cpp/test/Makefile.am | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/lib/cpp/test/Makefile.am b/lib/cpp/test/Makefile.am
index 8420357..399bebd 100755
--- a/lib/cpp/test/Makefile.am
+++ b/lib/cpp/test/Makefile.am
@@ -51,7 +51,6 @@ DebugProtoTest_extras.o: gen-cpp/DebugProtoTest_types.h
 libtestgencpp_la_LIBADD = $(top_builddir)/lib/cpp/libthrift.la

 noinst_PROGRAMS = Benchmark \
-	processor_test \
 	concurrency_test

 Benchmark_SOURCES = \
@@ -76,6 +75,12 @@ check_PROGRAMS = \
 	OpenSSLManualInitTest \
 	EnumTest

+if AMX_HAVE_LIBEVENT
+noinst_PROGRAMS +=
+	processor_test
+endif
+
+
 TESTS_ENVIRONMENT= \
 	BOOST_TEST_LOG_SINK=tests.xml \
 	BOOST_TEST_LOG_LEVEL=test_suite \
@@ -229,7 +234,7 @@ processor_test_LDADD = libprocessortest.la \
                        $(top_builddir)/lib/cpp/libthrift.la \
                        $(top_builddir)/lib/cpp/libthriftnb.la \
                        $(BOOST_LDFLAGS) \
-                       -levent \
+                       $(LIBEVENT_LIBS) \
                        -l:libboost_unit_test_framework.a

 OpenSSLManualInitTest_SOURCES = \
