--- breakpad-20150612/Makefile.in	2015-04-10 15:05:29.000000000 -0700
+++ breakpad-20150612-patched/Makefile.in	2015-06-16 21:02:01.675751605 -0700
@@ -219,13 +219,13 @@
 	$(srcdir)/breakpad-client.pc.in \
 	$(top_srcdir)/autotools/depcomp $(dist_doc_DATA) \
 	$(includec_HEADERS) $(am__includecl_HEADERS_DIST) \
-	$(am__includeclc_HEADERS_DIST) $(am__includeclh_HEADERS_DIST) \
-	$(am__includeclm_HEADERS_DIST) $(includegbc_HEADERS) \
-	$(am__includelss_HEADERS_DIST) $(includep_HEADERS) \
-	$(top_srcdir)/autotools/test-driver AUTHORS ChangeLog INSTALL \
-	NEWS README autotools/compile autotools/config.guess \
-	autotools/config.sub autotools/depcomp autotools/install-sh \
-	autotools/missing autotools/ltmain.sh \
+	$(am__includeclc_HEADERS_DIST) $(am__includeclcm_HEADERS_DIST) \
+	$(am__includeclh_HEADERS_DIST) $(am__includeclm_HEADERS_DIST) \
+	$(includegbc_HEADERS) $(am__includelss_HEADERS_DIST) \
+	$(includep_HEADERS) $(top_srcdir)/autotools/test-driver \
+	AUTHORS ChangeLog INSTALL NEWS README autotools/compile \
+	autotools/config.guess autotools/config.sub autotools/depcomp \
+	autotools/install-sh autotools/missing autotools/ltmain.sh \
 	$(top_srcdir)/autotools/compile \
 	$(top_srcdir)/autotools/config.guess \
 	$(top_srcdir)/autotools/config.sub \
@@ -272,9 +272,10 @@
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(bindir)" \
 	"$(DESTDIR)$(docdir)" "$(DESTDIR)$(pkgconfigdir)" \
 	"$(DESTDIR)$(includecdir)" "$(DESTDIR)$(includecldir)" \
-	"$(DESTDIR)$(includeclcdir)" "$(DESTDIR)$(includeclhdir)" \
-	"$(DESTDIR)$(includeclmdir)" "$(DESTDIR)$(includegbcdir)" \
-	"$(DESTDIR)$(includelssdir)" "$(DESTDIR)$(includepdir)"
+	"$(DESTDIR)$(includeclcdir)" "$(DESTDIR)$(includeclcmdir)" \
+	"$(DESTDIR)$(includeclhdir)" "$(DESTDIR)$(includeclmdir)" \
+	"$(DESTDIR)$(includegbcdir)" "$(DESTDIR)$(includelssdir)" \
+	"$(DESTDIR)$(includepdir)"
 LIBRARIES = $(lib_LIBRARIES) $(noinst_LIBRARIES)
 AR = ar
 ARFLAGS = cru
@@ -1564,15 +1565,18 @@
 am__includecl_HEADERS_DIST = $(top_srcdir)/src/common/linux/*.h
 am__includeclc_HEADERS_DIST =  \
 	$(top_srcdir)/src/client/linux/crash_generation/*.h
+am__includeclcm_HEADERS_DIST =  \
+	$(top_srcdir)/src/client/linux/dump_writer_common/*.h
 am__includeclh_HEADERS_DIST =  \
 	$(top_srcdir)/src/client/linux/handler/*.h
 am__includeclm_HEADERS_DIST =  \
 	$(top_srcdir)/src/client/linux/minidump_writer/*.h
 am__includelss_HEADERS_DIST = $(top_srcdir)/src/third_party/lss/*.h
 HEADERS = $(includec_HEADERS) $(includecl_HEADERS) \
-	$(includeclc_HEADERS) $(includeclh_HEADERS) \
-	$(includeclm_HEADERS) $(includegbc_HEADERS) \
-	$(includelss_HEADERS) $(includep_HEADERS)
+	$(includeclc_HEADERS) $(includeclcm_HEADERS) \
+	$(includeclh_HEADERS) $(includeclm_HEADERS) \
+	$(includegbc_HEADERS) $(includelss_HEADERS) \
+	$(includep_HEADERS)
 am__tagged_files = $(HEADERS) $(SOURCES) $(TAGS_FILES) $(LISP)
 # Read a list of newline-separated strings from the standard input,
 # and print each of them once, without duplicates.  Input order is
@@ -1917,6 +1921,8 @@
 @LINUX_HOST_TRUE@includeclh_HEADERS = $(top_srcdir)/src/client/linux/handler/*.h
 @LINUX_HOST_TRUE@includeclmdir = $(includedir)/$(PACKAGE)/client/linux/minidump_writer
 @LINUX_HOST_TRUE@includeclm_HEADERS = $(top_srcdir)/src/client/linux/minidump_writer/*.h
+@LINUX_HOST_TRUE@includeclcmdir = $(includedir)/$(PACKAGE)/client/linux/dump_writer_common
+@LINUX_HOST_TRUE@includeclcm_HEADERS = $(top_srcdir)/src/client/linux/dump_writer_common/*.h
 @LINUX_HOST_TRUE@includeclcdir = $(includedir)/$(PACKAGE)/client/linux/crash_generation
 @LINUX_HOST_TRUE@includeclc_HEADERS = $(top_srcdir)/src/client/linux/crash_generation/*.h
 @LINUX_HOST_TRUE@includelssdir = $(includedir)/$(PACKAGE)/third_party/lss
@@ -7117,6 +7123,27 @@
 	@list='$(includeclc_HEADERS)'; test -n "$(includeclcdir)" || list=; \
 	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
 	dir='$(DESTDIR)$(includeclcdir)'; $(am__uninstall_files_from_dir)
+install-includeclcmHEADERS: $(includeclcm_HEADERS)
+	@$(NORMAL_INSTALL)
+	@list='$(includeclcm_HEADERS)'; test -n "$(includeclcmdir)" || list=; \
+	if test -n "$$list"; then \
+	  echo " $(MKDIR_P) '$(DESTDIR)$(includeclcmdir)'"; \
+	  $(MKDIR_P) "$(DESTDIR)$(includeclcmdir)" || exit 1; \
+	fi; \
+	for p in $$list; do \
+	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
+	  echo "$$d$$p"; \
+	done | $(am__base_list) | \
+	while read files; do \
+	  echo " $(INSTALL_HEADER) $$files '$(DESTDIR)$(includeclcmdir)'"; \
+	  $(INSTALL_HEADER) $$files "$(DESTDIR)$(includeclcmdir)" || exit $$?; \
+	done
+
+uninstall-includeclcmHEADERS:
+	@$(NORMAL_UNINSTALL)
+	@list='$(includeclcm_HEADERS)'; test -n "$(includeclcmdir)" || list=; \
+	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
+	dir='$(DESTDIR)$(includeclcmdir)'; $(am__uninstall_files_from_dir)
 install-includeclhHEADERS: $(includeclh_HEADERS)
 	@$(NORMAL_INSTALL)
 	@list='$(includeclh_HEADERS)'; test -n "$(includeclhdir)" || list=; \
@@ -7860,7 +7887,7 @@
 all-am: Makefile $(LIBRARIES) $(PROGRAMS) $(SCRIPTS) $(DATA) \
 		$(HEADERS)
 installdirs:
-	for dir in "$(DESTDIR)$(libdir)" "$(DESTDIR)$(bindir)" "$(DESTDIR)$(docdir)" "$(DESTDIR)$(pkgconfigdir)" "$(DESTDIR)$(includecdir)" "$(DESTDIR)$(includecldir)" "$(DESTDIR)$(includeclcdir)" "$(DESTDIR)$(includeclhdir)" "$(DESTDIR)$(includeclmdir)" "$(DESTDIR)$(includegbcdir)" "$(DESTDIR)$(includelssdir)" "$(DESTDIR)$(includepdir)"; do \
+	for dir in "$(DESTDIR)$(libdir)" "$(DESTDIR)$(bindir)" "$(DESTDIR)$(docdir)" "$(DESTDIR)$(pkgconfigdir)" "$(DESTDIR)$(includecdir)" "$(DESTDIR)$(includecldir)" "$(DESTDIR)$(includeclcdir)" "$(DESTDIR)$(includeclcmdir)" "$(DESTDIR)$(includeclhdir)" "$(DESTDIR)$(includeclmdir)" "$(DESTDIR)$(includegbcdir)" "$(DESTDIR)$(includelssdir)" "$(DESTDIR)$(includepdir)"; do \
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
 	done
 install: install-am
@@ -7967,9 +7994,10 @@
 
 install-data-am: install-dist_docDATA install-includecHEADERS \
 	install-includeclHEADERS install-includeclcHEADERS \
-	install-includeclhHEADERS install-includeclmHEADERS \
-	install-includegbcHEADERS install-includelssHEADERS \
-	install-includepHEADERS install-pkgconfigDATA
+	install-includeclcmHEADERS install-includeclhHEADERS \
+	install-includeclmHEADERS install-includegbcHEADERS \
+	install-includelssHEADERS install-includepHEADERS \
+	install-pkgconfigDATA
 
 install-dvi: install-dvi-am
 
@@ -8018,10 +8046,11 @@
 
 uninstall-am: uninstall-binPROGRAMS uninstall-dist_docDATA \
 	uninstall-includecHEADERS uninstall-includeclHEADERS \
-	uninstall-includeclcHEADERS uninstall-includeclhHEADERS \
-	uninstall-includeclmHEADERS uninstall-includegbcHEADERS \
-	uninstall-includelssHEADERS uninstall-includepHEADERS \
-	uninstall-libLIBRARIES uninstall-pkgconfigDATA
+	uninstall-includeclcHEADERS uninstall-includeclcmHEADERS \
+	uninstall-includeclhHEADERS uninstall-includeclmHEADERS \
+	uninstall-includegbcHEADERS uninstall-includelssHEADERS \
+	uninstall-includepHEADERS uninstall-libLIBRARIES \
+	uninstall-pkgconfigDATA
 
 .MAKE: check-am install-am install-strip
 
@@ -8039,17 +8068,18 @@
 	install-dvi-am install-exec install-exec-am install-html \
 	install-html-am install-includecHEADERS \
 	install-includeclHEADERS install-includeclcHEADERS \
-	install-includeclhHEADERS install-includeclmHEADERS \
-	install-includegbcHEADERS install-includelssHEADERS \
-	install-includepHEADERS install-info install-info-am \
-	install-libLIBRARIES install-man install-pdf install-pdf-am \
-	install-pkgconfigDATA install-ps install-ps-am install-strip \
-	installcheck installcheck-am installdirs maintainer-clean \
-	maintainer-clean-generic mostlyclean mostlyclean-compile \
-	mostlyclean-generic pdf pdf-am ps ps-am recheck tags tags-am \
-	uninstall uninstall-am uninstall-binPROGRAMS \
-	uninstall-dist_docDATA uninstall-includecHEADERS \
-	uninstall-includeclHEADERS uninstall-includeclcHEADERS \
+	install-includeclcmHEADERS install-includeclhHEADERS \
+	install-includeclmHEADERS install-includegbcHEADERS \
+	install-includelssHEADERS install-includepHEADERS install-info \
+	install-info-am install-libLIBRARIES install-man install-pdf \
+	install-pdf-am install-pkgconfigDATA install-ps install-ps-am \
+	install-strip installcheck installcheck-am installdirs \
+	maintainer-clean maintainer-clean-generic mostlyclean \
+	mostlyclean-compile mostlyclean-generic pdf pdf-am ps ps-am \
+	recheck tags tags-am uninstall uninstall-am \
+	uninstall-binPROGRAMS uninstall-dist_docDATA \
+	uninstall-includecHEADERS uninstall-includeclHEADERS \
+	uninstall-includeclcHEADERS uninstall-includeclcmHEADERS \
 	uninstall-includeclhHEADERS uninstall-includeclmHEADERS \
 	uninstall-includegbcHEADERS uninstall-includelssHEADERS \
 	uninstall-includepHEADERS uninstall-libLIBRARIES \
